67 docs for template provider (#132)
notifirehq/notifire: docs/docs/community/create-provider.md

For a provider template you can copy one of our existing provider in the `providers` folder in the notifire project, make the relevant changes and create a PR against the monorepo.

## Description

> Providers allow us to handle message delivery over multiple channels. We have multiple providers for each channel (SMS, Email, Push, Direct, and orthers). To get started with adding a new provider lets look at setting up our repository. 

Reviewer: ```suggestion
Providers allow us to handle message delivery over multiple channels. We have multiple providers for each channel (SMS, Email, Push, Direct, and others). To get started with adding a new provider let's look at setting up our repository. 
```

## Initialization

To setup the repository, be sure to install the required packages. 

```zsh
yarn install
```

At the root of the project build the `core` package to get started.

```zsh
cd packages/core && yarn run build
```

## Snippets

After the project is initialized creating a new provider is achievable with the following.

```zsh
yarn run generate:provide
```

Choose the provider type.

```zsh
yarn run v1.22.17
$ npx hygen provider new
? What type of provider is this? … 
❯ EMAIL
  SMS
```

> For this example we will be selecting `EMAIL` as our provider type. The name for our provider will be exampleProvider.

Reviewer: ```suggestion
For this example, we will be selecting `EMAIL` as our provider type. The name for our provider will be exampleProvider.
```

```zsh
? Write the provider name camelCased: › exampleProvider
```

> Once our exampleProvider is generated we will need to begin working from within `/providers/exampleProvider` to begin adding our provider. Be sure to write test alongside your provider. See below for template examples for our `exampleProvider`.

Reviewer: ```suggestion
Once our exampleProvider is generated we will need to begin working from within `/providers/exampleProvider` to begin adding our provider. Be sure to write the test alongside your provider. See below for template examples for our `exampleProvider`.
```
